docs/content/v0.20/architecture.md

```diff
@@ -309,40 +310,20 @@ follows:
-              to a hashmap setting the name of the cluster as a key. Cluster
-              groups from different bindind policies are merged together.
-          - If any of the matched `BindingPolicy` has ` WantSingletonReportedState`
-            set to true, clusters are sorted in alphanumerical order and
-            only the first cluster is selected for delivery. Note that setting 
-            `WantSingletonReportedState` in one of the binding-policies that 
-            matches the object affects the behavior for all matching binding-policies, 
-            those with matching clusters and those w/o matching clusters.
-        - If there are no matching clusters, the worker returns without 
-       actions and is ready to process other events from the queue. 
-        - If there are matching clusters:
-          - Wraps the object into a ManifestWork
-       - Adds a label for each matched `BindingPolicy` to the ManifestWork that 
-         is used to track the `BindingPolicy` that caused the object to be
-         delivered to one or more clusters. The label contains both the 
-         `BindingPolicy` name (note that `BindingPolicy` is cluster-scoped) and the WDS name.
-         This way, when a binding-policy is deleted or updated it is possible to 
-         locate the associated ManifestWorks for deletion or label removal 
-         (if more than one label is present, as other binding-policies may “own” the object).
-       - For each cluster:
-         -  Sets the manifest namespace == name of the cluster
-         -  Uses the client connected to the ITS to do a server-side
-            apply patch of the manifest into the namespace.
-         - At this time there is only one field manager name for the server-side
-           apply (`kubestellar`) but the name should include also the WDS
-           name to allow detecting conflicts when two different field managers
-           try to patch a manifest with the same name.   
+        - Whether the object is a match or not, the worker notes the object as a match for the in-memory representation 
+       of the relevant `Binding`. If the noting of an object results in a change in the in-memory representation of the
+         `Binding`, the worker enqueues the latter for syncing.
+        - If a matched `BindingPolicy` has `WantSingletonReportedState` set to true (**see note below), the object is 
+       labeled with a special label in order to be able to track it for status reporting.
+        - If no matching `BindingPolicy` has `WantSingletonReportedState` set to true, the worker removes the label if
+       exists.
     - Worker returns and is ready to pick other keys from the queue.
 
+**WantSingletonReportedState**:
+currently it is the user's responsibility to make sure that a binding-policy that sets `WantSingletonReportedState` to
+true is not in conflict with other binding-policies that do the same, and that the binding-policy selects only a single
```

Comment:
Is "that do the same" an accurate limitation here? Will the implementation do the right thing if two BindingPolicy objects match a given workload object, where one BindingPolicy has `WantSingletonReportedState == true` and the other has it `false`?

Reply:
In that case, the one that has `WantSingletonReportedState` set to true will lead to singleton status reporting for the workload object (assuming it does not conflict with any other, and it selects a single cluster) - and the other one will have no effect on status, but will lead to getting the object delivered to all clusters it selects.

Reply:
@MikeSpreitzer the question of "will the implementation do the right thing" is problematic, cause we didn't define what is the right thing.
btw, if you ask me, the singleton status is a field for status collection and should not affect the distribution. 
so in that case I believe Maroon implemented it "the right way".

Reply:
@nirrozenbaum : my comment here is on documentation of the user's responsibility, my concern is that it is not an accurate reflection of the limitation of the current implementation.

If we are lucky, we will resolve #1734 soon and fix the implementation and documentation soon.

Reply:
@MikeSpreitzer you're right, the API documentation was changed previously during the review, I've updated it again to be aligned:

>
	// WantSingletonReportedState indicates that (a) the number of selected locations is intended
	// to be 1 and (b) the reported state of each downsynced object should be returned back to
	// the object in this space.
	// When this field is true, there should be no other BindingPolicy with `WantSingletonReportedState`
	// == true that matches any of the same workload objects.
	//
	// At the moment, it is the user's responsibility to comply to the above.

Reply:
The expected restriction scope is over the singleton-requiring bindingpolicies. I don't see a problem in the case of a binding-policy requiring singleton status reported over a set of objects, while selecting a single cluster, and a different binding-policy matching the same set of objects for distribution without requiring status reporting.

The problem is when there are multiple sources for **singleton** status.

Reply:
@MikeSpreitzer done